Kernel/Arch/i386/CPU.h

```diff
@@ -296,6 +296,10 @@ void load_task_register(u16 selector);
 #define sti() asm volatile("sti" :: \
                                : "memory")
 
+#define full_memory_barrier() __sync_synchronize()
+#define compiler_memory_barrier() asm volatile("" :: \
```

Comment:
We should just add the memory fences that are standardized to AK::Atomic.h, which will emit the appropriate instructions.

Reply:
I don't think I care about atomic operations (should I need to?), I do care that the CPU will not reorder memory accesses because it can result in undefined behavior in some cases.

Reply:
It's the same thing. The memory fences aren't really anything that have anything to do with atomic operations. It's all about reordering.

Reply:
https://www.modernescpp.com/index.php/fences-as-memory-barriers

Reply:
A full barrier is simply `memory_order_acq_rel`

Reply:
OK, thanks for the explanation :)

Reply:
I fixed the code to use `atomic_thread_fence(AK::MemoryOrder::memory_order_acq_rel)`, I hope it's enough for ensuring memory order :)

Reply:
Actually, you probably also need to pair it with `atomic_signal_fence` to prevent compiler reordering.

Reply:
But we don't have such function in AK/Atomic.h, so I think we should add a new function for it and pair these functions together in one `full_memory_barrier` function in Atomic.h :)